spec/classes/init_spec.rb

```diff
@@ -51,7 +51,7 @@
 
   shared_examples 'below range' do
     let(:params) { { param_sym => below_range } }
-    let(:error_message) { %r{expects a value of type Undef, Integer} }
+    let(:error_message) { %r{expects a value of type Undef, Integer\[#{range[0]}, #{range[1]}\], or Boolean} }
```

Comment:
Whats happening here?

Reply:
This is a stricter check of the error message that is expected to be returned. For example:
```
let(:error_message) { %r{expects a value of type Undef, Integer} } 
```
matched to any error message that contained those characters, were as the actually error message is returned in the format "expects a value of undef or Integer[2, 15], or Boolean, got [1, 1]"

As this is a shared example, it was necessary to interpolate the error message with the valid range passed from each test case, which is why each test case now has a variable like this:
```
        let(:range) { [2, 15] }
``` 

In this example [2,15] is the valid range of values that can be passed, and therefore [1,1] is below range and is invalid (which is the test case) :-)

Reply:
Top stuff, gotcha!